### ISSUE TITLE
NoMethodError: undefined method `empty?' for nil:NilClass

### ISSUE THREAD

**Issue by feixionglee** [guilhermesad/rspotify#123]
if I do not make authenticate at first, it raise this error, 

```
NoMethodError: undefined method `empty?' for nil:NilClass
from **/lib/rspotify/connection.rb:75:in `send_request'
```

my gem version is the latest 1.22.1

**Comment by feixionglee** [comment]
Is it better if  raise Unauthorized exception?

**Comment by CGrijalva90** [comment]
I have a similar issue, I tried messing around with it without authenticating a user with the following code:

```
post '/music' do
  search =  RSpotify::Artist.search(params[:search])
    @artist = search.first
    @image = @artist.images.first["url"]
    @top_tracks = @artist.top_tracks(:US)
    albums = @artist.albums
    @titles = []
    albums.each { |album| @titles << album.name }
    @titles.uniq!
    @related_artists = @artist.related_artists
    erb :'/music/show'
end
```

Was working a few weeks ago, now i'm getting:

```
NoMethodError: undefined method `empty?' for nil:NilClass
from **/lib/rspotify/connection.rb:75:in `send_request'
```

**Comment by guilhermesad** [comment]
Solved in #170.
